File: app/src/main/java/com/zyh/wanandroid/di/component/ActivityComponent.java
package com.zyh.wanandroid.di.component;

import com.zyh.wanandroid.di.module.ActivityModule;
import com.zyh.wanandroid.di.scope.PerActivity;
import com.zyh.wanandroid.ui.main.MainActivity;
import com.zyh.wanandroid.ui.splash.SplashActivity;

import dagger.Component;

/**
 * author : zyh
 * Date : 2019/1/28
 * Description :
 */
@PerActivity
@Component(dependencies = ApiComponent.class,modules = ActivityModule.class)
public interface ActivityComponent {
    void inject(SplashActivity splashActivity);
    void inject(MainActivity mainActivity);
}

File: app/src/main/java/com/zyh/wanandroid/di/component/ActivityComponent.kt
package com.zyh.wanandroid.di.component

import com.zyh.wanandroid.di.module.ActivityModule
import com.zyh.wanandroid.di.scope.PerActivity
import dagger.Component

/**
 * author : zyh
 * Date : 2018/11/23
 * Description :
 */
@PerActivity
@Component(dependencies = arrayOf(ApiComponent ::class),modules = arrayOf(ActivityModule::class))
public interface ActivityComponent{

}